src/engine/renderer/tr_main.cpp

```diff
@@ -2130,7 +2132,22 @@ static void R_SortDrawSurfs()
 	}
 
 	if ( glConfig2.materialSystemAvailable ) {
-		materialSystem.AddPortalSurfaces();
+		if ( tr.viewParms.portalLevel == 0 ) {
+			materialSystem.AddPortalSurfaces();
+			currentView = 0;
+		} else {
+			currentView++;
+		}
+
+		for ( uint32_t i = 0; i < portalStack[currentView].count; i++ ) {
+			uint32_t viewID = portalStack[currentView].views[i];
+			if ( viewID == 0 ) {
+				break;
+			}
+
+			R_MirrorViewBySurface( portalStack[portalStack[currentView].views[i]].drawSurf );
```

Comment:
R_MirrorViewBySurface was already called a view lines above, and it does the entire portal/RenderView mutual recursion, so calling it again seems odd.